src/state.rs

```diff
@@ -38,25 +38,72 @@ impl OPRFServer {
 }
 
 /// Advance to the next epoch on a timer
-/// This can be invoked as a background task to handle
-/// epoch advance and key rotation according to the
-/// given Config.
+/// This can be invoked as a background task to handle epoch
+/// advance and key rotation according to the given Config.
 #[instrument(skip_all)]
 pub async fn epoch_loop(state: OPRFState, config: &Config) {
+    let epochs = config.first_epoch..=config.last_epoch;
+
     let interval =
         std::time::Duration::from_secs(config.epoch_seconds.into());
     info!("rotating epoch every {} seconds", interval.as_secs());
 
-    let epochs = config.first_epoch..=config.last_epoch;
+    let start_time = time::OffsetDateTime::now_utc();
+    // Epoch base_time comes from a config argument if given,
+    // otherwise use start_time.
+    let base_time = config.epoch_base_time.unwrap_or(start_time);
+    info!(
+        "epoch base time {}",
```

Comment:
nit: maybe the template should be something like "epoch base time = {}" or "epoch base time is now {}" for cleanliness